docs/sql-programming-guide.md

```diff
@@ -1640,6 +1640,129 @@ Configuration of Hive is done by placing your `hive-site.xml`, `core-site.xml` a
 You may run `./bin/spark-sql --help` for a complete list of all available
 options.
 
+# PySpark Usage Guide for Pandas with Arrow
+
+## Arrow in Spark
+
+Apache Arrow is an in-memory columnar data format that is used in Spark to efficiently transfer
+data between JVM and Python processes. This currently is most beneficial to Python users that
+work with Pandas/NumPy data. Its usage is not automatic and might require some minor
+changes to configuration or code to take full advantage and ensure compatibility. This guide will
+give a high-level description of how to use Arrow in Spark and highlight any differences when
+working with Arrow-enabled data.
+
+### Ensure PyArrow Installed
+
+If you install PySpark using pip, then PyArrow can be brought in as an extra dependency of the
+SQL module with the command `pip install pyspark[sql]`. Otherwise, you must ensure that PyArrow
+is installed and available on all cluster nodes. The current supported version is 0.8.0.
+You can install using pip or conda from the conda-forge channel. See PyArrow
+[installation](https://arrow.apache.org/docs/python/install.html) for details.
+
+## Enabling for Conversion to/from Pandas
+
+Arrow is available as an optimization when converting a Spark DataFrame to Pandas using the call
+`toPandas()` and when creating a Spark DataFrame from Pandas with `createDataFrame(pandas_df)`.
+To use Arrow when executing these calls, users need to first set the Spark configuration
+'spark.sql.execution.arrow.enabled' to 'true'. This is disabled by default.
```

Comment:
is `'spark.sql.execution.arrow.enabled' to 'true'` only for toPandas and does not affect `pandas_udf`?

As a side note, I feel we should discourage the use of toPandas since it will force everything to the Driver...

Reply:
> is 'spark.sql.execution.arrow.enabled' to 'true' only for toPandas and does not affect pandas_udf?

Yup, it is. It's for both `toPandas` and `createDataFrame`.

Reply:
then in doc and example I think it would be great to pull that outside of toPandas to avoid confusion?

Reply:
+1 for any way to explicitly express it.

Reply:
>  I feel we should discourage the use of toPandas

I am not sure that's necessary. I think it's reasonable to down-sample/aggregate data in Spark and use `toPandas()` to bring small data to local and analyze further or visualize. Maybe instead we should discourage use of `toPandas` with large amounts of data?

Reply:
Yea, I am actually slightly with you more but let's leave that discussion out here.

Reply:
I wouldn't hurt to remind the user that this collects all data to the driver, I'll try to work that in

Reply:
I added a note for this